3.7/ubuntu/Dockerfile

```diff
@@ -0,0 +1,271 @@
+    curl --verbose --location --silent --fail --fail-early --output $RABBITMQ_PATH.tar.xz $RABBITMQ_SOURCE_URL && \
+    curl --verbose --location --silent --fail --fail-early --output $RABBITMQ_PATH.tar.xz.asc $RABBITMQ_SOURCE_URL.asc && \
+    gpg --batch --keyserver $PGP_KEYSERVER --recv-keys "$RABBITMQ_PGP_KEY_ID" && \
+    gpg --batch --verify $RABBITMQ_PATH.tar.xz.asc $RABBITMQ_PATH.tar.xz && \
+    tar -xvf $RABBITMQ_PATH.tar.xz --directory $RABBITMQ_INSTALL_DIR --strip-components=1
+
+# Do not default SYS_PREFIX to RABBITMQ_HOME, leave it empty
+RUN sed -i 's/^SYS_PREFIX=.*/SYS_PREFIX=/' $RABBITMQ_INSTALL_DIR/sbin/rabbitmq-defaults
+
+# Start with the unmodified release image
+FROM release
+# Duplicate ARGs since they are not preserved between stages in multi-stage builds
+ARG OPENSSL_INSTALL_DIR=/opt/openssl
+ARG OPENSSL_CONFIG_DIR=/usr/local/ssl
+ARG OTP_INSTALL_DIR=/opt/otp
+ARG RABBITMQ_INSTALL_DIR=/opt/rabbitmq
+
+ARG RABBITMQ_DATA_DIR=/var/lib/rabbitmq
+ARG RABBITMQ_LOG_DIR=/var/log/rabbitmq
+ARG RABBITMQ_CONF_DIR=/etc/rabbitmq
+ARG RABBITMQ_TMP_SSL_DIR=/tmp/rabbitmq-ssl
+# Only copy the build artefacts required to run RabbitMQ
+COPY --from=builder $OPENSSL_INSTALL_DIR $OPENSSL_INSTALL_DIR
+COPY --from=builder $OPENSSL_CONFIG_DIR $OPENSSL_CONFIG_DIR
+COPY --from=builder $OTP_INSTALL_DIR $OTP_INSTALL_DIR
+COPY --from=builder $RABBITMQ_INSTALL_DIR $RABBITMQ_INSTALL_DIR
+
+# Add OpenSSL to PATH
+ENV PATH=$OPENSSL_INSTALL_DIR/bin:$PATH LD_LIBRARY_PATH=$OPENSSL_INSTALL_DIR/lib
+# Ensure OpenSSL works by listing all commands
+RUN openssl help
+
+# Add Erlang/OTP to PATH
+ENV PATH=$OTP_INSTALL_DIR/bin:$PATH
+# Ensure Erlang/OTP works by listing all loaded modules
+RUN erl -noshell -eval 'io:format("~p~n", [code:all_loaded()]), init:stop().'
+
+# Create rabbitmq system user & group, fix permissions & allow root user to connect to the RabbitMQ Erlang VM
+RUN groupadd --system rabbitmq && \
+    useradd --system --home-dir $RABBITMQ_DATA_DIR --create-home --gid rabbitmq rabbitmq && \
+    mkdir -p $RABBITMQ_CONF_DIR $RABBITMQ_LOG_DIR $RABBITMQ_DATA_DIR $RABBITMQ_TMP_SSL_DIR $RABBITMQ_TMP_SSL_DIR && \
+    chown -fR rabbitmq:rabbitmq $RABBITMQ_INSTALL_DIR $RABBITMQ_CONF_DIR $RABBITMQ_LOG_DIR $RABBITMQ_DATA_DIR && \
+    ln -sf $RABBITMQ_DATA_DIR/.erlang.cookie /root/.erlang.cookie
+
+# TODO @tianon why is this necessary?
+# RUN ln -sf "/usr/lib/rabbitmq/lib/rabbitmq_server-$RABBITMQ_VERSION/plugins" /plugins
+
+# Hint that the config, log & data (a.k.a. home dir) dirs should be separate volumes
+VOLUME $RABBITMQ_CONF_DIR
+VOLUME $RABBITMQ_LOG_DIR
+VOLUME $RABBITMQ_DATA_DIR
+
+# Add RabbitMQ to PATH, send all logs to TTY
+# TODO @tianon is sbin a problem? Notice that I'm installing generic-unix, not the Debian package
+ENV PATH=$RABBITMQ_INSTALL_DIR/sbin:$PATH RABBITMQ_LOGS=-
+
+# Improve default locale: locale -a
+# https://docs.docker.com/samples/library/ubuntu/#locales
+ENV LANG=C.UTF-8 LANGUAGE=C.UTF-8 LC_ALL=C.UTF-8
```

Comment:
What's the rationale for setting all three of these?  Our current `Dockerfile` only does `LANG` which seems to have the desired effect:

https://github.com/docker-library/rabbitmq/blob/b3faf3d1e1c3b5a13b32d98babeba72c394d7ef4/3.7/debian/Dockerfile#L125-L126

(Which also has a comment that explains the "why change this?" rationale a little better than that cryptic Ubuntu link :sweat_smile:)

Reply:
Does 668df358b8a5682e2d1230a3b3ff4bda7301e42d address it?